### PR TITLE
Add last known good rendering of card in error state to stack item

### PR DESCRIPTION
**Pull request by habdelra**
This PR adds the last known good rendering of a card in an error state for a stack item. In the case there is no last known good HTML, we'll show a generic message.

Note that this does not yet support live updates into and out of an error state. that will come in a subsequent PR.

error

This is the view for a card without a last known good state:

Screenshot from 2024-11-27 13-14-37

### CODE REVIEW

File: packages/host/app/components/operator-mode/interact-submode.gts [hunk 1]
```diff
@@ -341,6 +341,11 @@ export default class InteractSubmode extends Component<Signature> {
...
 
   private close = task(async (item: StackItem) => {
+    if (item.cardError) {
+      this.operatorModeStateService.trimItemsFromStack(item);
+      return;
+    }
+
     let { card, request } = item;
     // close the item first so user doesn't have to wait for the save to complete
     this.operatorModeStateService.trimItemsFromStack(item);
```
Comment (lukemelia) [inline packages/host/app/components/operator-mode/interact-submode.gts]:
```suggestion
    // close the item first so user doesn't have to wait for the save to complete
    this.operatorModeStateService.trimItemsFromStack(item);
    if (item.cardError) {
      return;
    }

    let { card, request } = item;
```

File: packages/host/app/resources/stack-backgrounds.ts [hunk 2]
```diff
@@ -46,6 +46,9 @@ export class StackBackgroundsResource extends Resource<Args> {
         }
         let bottomMostStackItem = stack[0];
         await bottomMostStackItem.ready;
+        if (bottomMostStackItem.cardError) {
+          return undefined;
```
Comment (lukemelia) [inline packages/host/app/resources/stack-backgrounds.ts]: Is it impossible to know what realm an error doc is from?

**Review by lukemelia — APPROVED** [review]

File: packages/host/app/resources/stack-backgrounds.ts (hunk 2 above)
Comment (habdelra) [reply packages/host/app/resources/stack-backgrounds.ts]: Hmm, I’ll look into it. Presumably some realm is returning the error response. Perhaps it’s a header? If so we’ll have to refactor how the stack item calculates the realm, as right now I does so from a running card.

Comment (lukemelia) [reply packages/host/app/resources/stack-backgrounds.ts]: Maybe it makes sense for the error doc to carry the realm?

Comment (habdelra) [reply packages/host/app/resources/stack-backgrounds.ts]: Good idea

Comment (habdelra) [reply packages/host/app/resources/stack-backgrounds.ts]: addressed in https://github.com/cardstack/boxel/pull/1834/commits/64e950c562812baca7c245484cb49a506f2f7fc0

**Review by lukemelia — APPROVED** [review]

**Pull request merged**